tiny_dnn/core/kernels/fully_connected_op_internal.h

```diff
@@ -12,59 +12,67 @@
 namespace tiny_dnn {
 namespace kernels {
 
-inline void fully_connected_op_internal(const tensor_t &in_data,
-                                        const vec_t &W,
-                                        const vec_t &bias,
-                                        tensor_t &out_data,
+template <typename S1, typename S2, typename S3, typename S4>
+inline void fully_connected_op_internal(const Tensor<float_t, S1> &in_data,
+                                        const Tensor<float_t, S2> &weights,
+                                        const Tensor<float_t, S3> &bias,
+                                        Tensor<float_t, S4> &out_data,
                                         const fully_params &params,
                                         const bool layer_parallelize) {
-  for_i(layer_parallelize, in_data.size(), [&](int sample) {
-    const vec_t &in = in_data[sample];
-    vec_t &out      = out_data[sample];
-
+  for_i(layer_parallelize, in_data.shape()[0], [&](int sample) {
```

Comment:
Maybe not now but this is just a matrix multiplication, we could take advantage​ here of xtensor-blas